src/components/sidebar/right-sidebar/RightSidebar.tsx

```diff
@@ -7,12 +8,113 @@ import "./RightSidebar.css"
 import Translate from "../../../utils/translate/translateService";
 import { Link } from "react-router-dom";
 
+import * as moment from 'moment';
+import 'react-dates/initialize';
+import { DateRangePicker, SingleDatePicker, DayPickerRangeController } from 'react-dates';
+import 'react-dates/lib/css/_datepicker.css';
+
 export default function RightSidebar() {
+  const [state, dispatch] = useContext(AppContext);
+
+  const [startDate, setStartDate] = useState(null);
+  const [endDate, setEndDate] = useState(null);
+  const [focusedInput, setFocusedInput] = useState("startDate");
+
+  useEffect(() => {
+    if(state.airtable_records) {
+      const orderedDates = state.airtable_records
+      .filter(o => o.publish_date !== null && o.publish_date !== undefined)
+      .map(o => o.publish_date instanceof Array ? Date.parse(o.publish_date[0] as string) : Date.parse(o.publish_date as string))
+      .sort((a, b) => a - b);
+
+      const startDate = orderedDates[0];
+      setStartDate(moment(startDate));
+      setEndDate(moment());
+    }
+  }, [state.airtable_records])
+
+  const onDatesChange = (range: object) => {
+    console.log(range)
+    const start = range.startDate;
+    const end = range.endDate;
+    if (moment.isMoment(start)) {
+      setStartDate(start);
+    }
+
+    if (moment.isMoment(end)) {
+      setEndDate(end);
+    }
+
+    if (start != null && end != null) {
+      const values = [start.valueOf(), end.valueOf()]
+      dispatch({
+        type: 'UPDATE_FILTER',
+        payload: {
+          filter_type: 'publish_date',
+          filter_value: values
+        }
+      })
+    }
+  }
+
+  const onFocusChange = (focus: object) => {
+    setFocusedInput(focus);
+  }
+
+  function isAfterDay(a, b) {
+    if (!moment.isMoment(a) || !moment.isMoment(b)) return false;
+    return !isBeforeDay(a, b) && !isSameDay(a, b);
+  }
+
+  function isBeforeDay(a, b) {

```

Comment:
nitpick: Could be nice to make variable names a little more descriptive. 

We could also move this functionality to a separate file specifically for the datepicker just to keep things modular.

Reply:
I think for these pure helper functions that are just comparing two moments, a and b seem reasonable but I totally get the instinctual reluctance of seeing single letter vars haha